depends/patches/qt/utc_from_string_no_optimize.patch

```diff
@@ -0,0 +1,84 @@
+Modify optimisation flags for various functions.
+This fixes non-determinism issues in the asm produced for
+these function when cross-compiling on x86_64 and aarch64 for
+the arm64-apple-darwin HOST.
+
+--- a/qtbase/src/corelib/itemmodels/qitemselectionmodel.cpp
++++ b/qtbase/src/corelib/itemmodels/qitemselectionmodel.cpp
+@@ -1078,9 +1078,9 @@ void QItemSelectionModelPrivate::_q_layoutChanged(const QList<QPersistentModelIn
+ 
+     if (hint != QAbstractItemModel::VerticalSortHint) {
+         // sort the "new" selection, as preparation for merging
+-        std::stable_sort(savedPersistentIndexes.begin(), savedPersistentIndexes.end(),
++        std::sort(savedPersistentIndexes.begin(), savedPersistentIndexes.end(),
+                          qt_PersistentModelIndexLessThan);
+-        std::stable_sort(savedPersistentCurrentIndexes.begin(), savedPersistentCurrentIndexes.end(),
++        std::sort(savedPersistentCurrentIndexes.begin(), savedPersistentCurrentIndexes.end(),
+                          qt_PersistentModelIndexLessThan);
+ 
+         // update the selection by merging the individual indexes
+@@ -1092,8 +1092,8 @@ void QItemSelectionModelPrivate::_q_layoutChanged(const QList<QPersistentModelIn
+         savedPersistentCurrentIndexes.clear();
+     } else {
+         // sort the "new" selection, as preparation for merging
+-        std::stable_sort(savedPersistentRowLengths.begin(), savedPersistentRowLengths.end());
+-        std::stable_sort(savedPersistentCurrentRowLengths.begin(), savedPersistentCurrentRowLengths.end());
++        std::sort(savedPersistentRowLengths.begin(), savedPersistentRowLengths.end());
++        std::sort(savedPersistentCurrentRowLengths.begin(), savedPersistentCurrentRowLengths.end());
+ 
+         // update the selection by merging the individual indexes
+         ranges = mergeRowLengths(savedPersistentRowLengths);
+
+--- a/qtbase/src/corelib/itemmodels/qitemselectionmodel_p.h
++++ b/qtbase/src/corelib/itemmodels/qitemselectionmodel_p.h
+@@ -76,7 +76,7 @@ public:
+     void _q_rowsAboutToBeInserted(const QModelIndex &parent, int start, int end);
+     void _q_columnsAboutToBeInserted(const QModelIndex &parent, int start, int end);
+     void _q_layoutAboutToBeChanged(const QList<QPersistentModelIndex> &parents = QList<QPersistentModelIndex>(), QAbstractItemModel::LayoutChangeHint hint = QAbstractItemModel::NoLayoutChangeHint);
+-    void _q_layoutChanged(const QList<QPersistentModelIndex> &parents = QList<QPersistentModelIndex>(), QAbstractItemModel::LayoutChangeHint hint = QAbstractItemModel::NoLayoutChangeHint);
++    __attribute__ ((optnone)) void _q_layoutChanged(const QList<QPersistentModelIndex> &parents = QList<QPersistentModelIndex>(), QAbstractItemModel::LayoutChangeHint hint = QAbstractItemModel::NoLayoutChangeHint);
```

Comment:
b335710782c2545e6eeed67b5e1763c07eab26b0

Being a clang-specific attribute it should be properly gated to do not produce `warning: 'optnone' attribute directive ignored [-Wattributes]`.

However, such warnings are harmless for now.

Reply:
The Qt build already produces heaps of other irrelevant warning output, so I'm not sure this makes any real difference? Properly gateing this would also just increase the patch size, and make rebasing them harder?

This same thing also already occurs for other qt patches, i.e fast_fixed_dtoa_no_optimize, where clang already warns about ignoring the pragma usage.